lib/galaxy/util/oset.py

```diff
@@ -2,61 +2,70 @@
 Ordered set implementation from https://code.activestate.com/recipes/576694/
 """
 from collections.abc import MutableSet
+from typing import 
+from typing import (
+    Any,
+    Dict,
+    List,
+    TYPE_CHECKING,
+)
 
+if TYPE_CHECKING:
+    from bioblend.galaxy import GalaxyInstance
 
 class OrderedSet(MutableSet):
-    def __init__(self, iterable=None):
+    def __init__(self, iterable=None) -> List[Dict[str, Any]]:
         self.end = end = []
         end += [None, end, end]  # sentinel node for doubly linked list
         self.map = {}  # key --> [key, prev, next]
         if iterable is not None:
             self |= iterable
 
-    def __len__(self):
+    def __len__(self) -> Dict[str, Any]:
         return len(self.map)
 
-    def __contains__(self, key):
+    def __contains__(self, key) -> Dict[str, Any]:
         return key in self.map
 
-    def add(self, key):
+    def add(self, key) -> Dict[str, Any]:
         if key not in self.map:
             end = self.end
             curr = end[1]
             curr[2] = end[1] = self.map[key] = [key, curr, end]
 
-    def discard(self, key):
+    def discard(self, key) -> Dict[str, Any]:
         if key in self.map:
             key, prev, next = self.map.pop(key)
             prev[2] = next
             next[1] = prev
 
-    def __iter__(self):
+    def __iter__(self) -> Dict[str, Any]:
```

Comment:
all these are special (also None as the return type); they emulate python's syntax for container handling, `len(foo)`, etc. (see python's data model: https://docs.python.org/3/reference/datamodel.html#object.__iter__)